### PR TITLE
Add support for setting keyboard cursor shape

### PR DESCRIPTION
**Pull request by f2404**
Required for https://github.com/lxde/qterminal/issues/151

### CODE REVIEW

File: lib/qtermwidget.h [hunk 1]
```diff
@@ -41,6 +41,21 @@ class QTermWidget : public QWidget {
         ScrollBarRight=2
     };
 
+    enum KeyboardCursorShape {
```
Comment (paulolieuthier) [inline lib/qtermwidget.h]: Why duplicate the enum?

Comment (f2404) [inline lib/qtermwidget.h]: Because `qtermwidget.h` is the header exposed to the app, and the app should use the enum.

Comment (f2404) [inline lib/qtermwidget.h]: Do you suggest moving the enum declaration from `TerminalDisplay.h` to `qtermwidget.h`?
`enum ScrollBarPosition` is also duplicated.

Comment (paulolieuthier) [inline lib/qtermwidget.h]: Yeah, duplication should be avoided.

Comment (f2404) [inline lib/qtermwidget.h]: Yes, I agree. Will remove it in next commit.

**Comment by f2404** [comment]
`enum ScrollBarPosition` and `enum KeyboardCursorShape` are now defined in `qtermwidget.h`.

**Pull request merged**